pkg/network/node/ovscontroller.go

```diff
@@ -46,6 +48,41 @@ func (oc *ovsController) getVersionNote() string {
 	return fmt.Sprintf("%02X.%02X", oc.pluginId, ruleVersion)
 }
 
+func (oc *ovsController) updateOVSMetrics() {
+	flows, err := oc.ovs.DumpFlows()
```

Comment:
How expensive is this?

Reply:
Can this contend?

Reply:
@smarterclayton All this stuff could be expensive.  Which is why I'm not quite sure where to put it.

These things are great for debugging, we *definitely* want them when things go wrong.  But how do we only eat the cost when things go wrong or are about to go wrong, instead of all the time?  I can't seem to see anything with prometheus that lets us do that.

Reply:
To expand on this.  There are things (stale OVS flows, ARP cache size, free pod IPs) that indicate node health and when they reach certain levels we would consider the node close-to-unhealthy.  But they are also expensive to gather.  Hence the timer suggestion, rather than every sandbox operation.

Think of these things like PLEG; it has to 'docker ps' then 'docker inspect' every single container even if it's not a running kube pod, because that helps indicate resource leaks and enables garbage collection.  Expensive operation, but necessary for system health checking.

The other things (sandbox setup/teardown latency, etc) that are not expensive.

Reply:
I like the background gathering... how bad would it be if we did this every 5m?  Or would we need more frequent data to make good use of it?

Reply:
@knobunc not too bad, I'll implement this.  It'll delay the results somewhat, but likely better performance-wise than every pod action.

Reply:
> Can this contend?

No, it just calls out to ovs-ofctl, which does its own locking

Reply:
Generally we scrape every minute so you would usually use the collector pattern to do that.  If you can easily swap to that model it's going to work best for the "calculate something expensive infrequently pattern".  The haproxy scraper is an example of this in pkg/router/metrics/haproxy - the metrics are demand collected.  I think the hat might be sufficient for your case and is also pretty flexible.

Reply:
That's not a requirement for 3.7 but is probably easier because we can always back off collections.

Re expensive is it more or less expensive than walking an entire cgroup hierarchy on disk or iptables-save?

Reply:
Can you open a 3.8 follow up to switch to the collector pattern rather than goroutine loop?